engine/src/main/java/org/terasology/engine/entitySystem/entity/EntityBuilder.java

```diff
@@ -151,9 +151,10 @@ public boolean hasAllComponents(List<Class<? extends Component>> filterComponent
         return component;
     }
 
-    public void addComponents(Iterable<? extends Component> components) {
-        components = (components == null) ? Collections.EMPTY_LIST : components;
-        components.forEach(this::addComponent);
+    public void addComponents(Iterable<? extends Component> maybeComponents) {
+        Optional.ofNullable(maybeComponents).ifPresent(comps -> {
+            comps.forEach(this::addComponent);
+        });
```

Comment:
While `maybeComponents` makes sense for the parameter, I don't like that we have to use `comps` because `components` would hide a name... :confused: